inc/audit_log_to_cloudwatch/class-cloudwatch-driver.php

```diff
@@ -148,14 +148,32 @@ public function get_records( $args ) {
 		}
 
 		$results = [ 'status' => 'Running' ];
+		$start_searching_time = time();
 		while ( ! in_array( $results['status'], [ 'Failed', 'Cancelled', 'Complete' ], true ) ) {
+
 			// Limit how fast we poll CloudWatch via calls to getQueryResults.
 			// Queries take at a minimum 1 second, so we `sleep` before even
 			// making the first call.
 			sleep( 1 );
 			$results = CloudWatch_Logs\cloudwatch_logs_client()->getQueryResults([
 				'queryId' => $query['queryId'],
 			] );
+
+			$time_taken = time() - $start_searching_time;
+			// If we are looking for most recent results, we can short-circuit once we have enough
+			// in the results array, even if the query has not completed searching all records.
+			// We only trigger this behaviour at a minimum of 15 seconds, which gives the query time to
+			// complete as usual, which will mean that the total found results is correctly set.
+			// Short-circuiting the query early means the total counts will not be accurate, so we fix to 10k.
+			if ( $order === 'desc' && $results['status'] === 'Running' && count( $results['results'] ) === $limit && $time_taken > 15 ) {
+				$results['statistics']['recordsMatched'] = 10000;
```

Comment:
Not sure I follow why this number, and whether it's useful. Could we show the current known results count but indicate that there might be more instead?

Reply:
Yeah.. and no. Right now this is just a driver, so we have to return _something_. We could also modify the UI, but that's beyond what a driver can currently do.

FWIW this is not a new change, we already show 10k whenever there is more results than 10k.